common/lib/dependabot/errors.rb

```diff
@@ -376,23 +381,24 @@ def initialize(tool_name, detected_version, supported_versions)
   class DependencyFileNotFound < DependabotError
     extend T::Sig
 
-    sig { returns(String) }
+    sig { returns(T.nilable(String)) }
     attr_reader :file_path
 
+    sig { params(file_path: T.nilable(String), msg: T.nilable(String)).void }
     def initialize(file_path, msg = nil)
       @file_path = file_path
       super(msg || "#{file_path} not found")
     end
 
-    sig { returns(String) }
+    sig { returns(T.nilable(String)) }
     def file_name
-      T.must(file_path.split("/").last)
+      file_path&.split("/")&.last
     end
 
-    sig { returns(String) }
+    sig { returns(T.nilable(String)) }
     def directory
       # Directory should always start with a `/`
-      T.must(file_path.split("/")[0..-2]).join("/").sub(%r{^/*}, "/")
+      file_path&.split("/")&.[](0..-2)&.join("/")&.sub(%r{^/*}, "/")
```

Comment:
I'd rather split this into two several lines. It feels very obfuscated. Would

```ruby
return unless file_path

file_path.split("/")[0..-2].join("/").sub(%r{^/*}, "/")
```

work?

Reply:
Yes, but unfortunately, I have to wrap the 2nd line in `T.must`